meteor/sofie-server/data-access/repositories/test/mongo-rundown-playlist-repository.spec.ts

```diff
@@ -0,0 +1,144 @@
+		}
+		if (mongoServer) {
+			await mongoServer.stop()
+		}
+	})
+
+	describe('getRundown', () => {
+		it('has an activationId, return an active rundown', async () => {
+			const rundownId: string = 'activatedRundownId'
+			const testee: MongoRundownPlaylistRepository = await createTesteeForRundown(true, rundownId)
+			const result: Rundown = await testee.getRundown(rundownId)
+
+			expect(result.getActiveStatus()).toBe(true)
+		})
+		it('does not have an activationId, return deactivated rundown', async () => {
+			const rundownId: string = 'deactivatedRundownId'
+			const testee: MongoRundownPlaylistRepository = await createTesteeForRundown(false, rundownId)
+			const result: Rundown = await testee.getRundown(rundownId)
+
+			expect(result.getActiveStatus()).toBe(false)
+		})
+	})
+	describe('getBasicRundowns', () => {
+		it('returns two rundowns, one is active', async () => {
+			const firstRundownId: string = 'firstActivatedRundownId'
+			const secondRundownId: string = 'secondActivatedRundownId'
+			const testee: MongoRundownPlaylistRepository = await createTesteeForTwoRundowns(
+				true,
+				firstRundownId,
+				secondRundownId
+			)
+			const result: Rundown[] = await testee.getBasicRundowns()
+
+			expect(result[0].getActiveStatus()).toBe(true)
+		})
+		it('returns two rundowns, none are active', async () => {
+			const firstRundownId: string = 'firstDeactivatedRundownId'
+			const secondRundownId: string = 'secondDeactivatedRundownId'
+			const testee: MongoRundownPlaylistRepository = await createTesteeForTwoRundowns(
+				false,
+				firstRundownId,
+				secondRundownId
+			)
+			const result: Rundown[] = await testee.getBasicRundowns()
+
+			expect(result[0].getActiveStatus()).toBe(false)
+		})
+	})
+
+	async function createTesteeForRundown(
+		includeActiveRundown: boolean,
+		rundownId: string
+	): Promise<MongoRundownPlaylistRepository> {
+		const db = client.db(mongoServer.instanceInfo!.dbName)
+
+		const rundownName: string = 'INEWS_QUEUE_PATH'
+		const playlistExternalId: string = 'INEWS_QUEUE_PATH'
+
+		const rundown: Rundown = new Rundown({ id: rundownId, name: rundownName } as RundownInterface)
+
+		if (includeActiveRundown) {
+			await db.collection('rundownPlaylists').insertOne({
+				externalId: playlistExternalId,
+				activationId: 'activated',
+			})
+		} else {
+			await db.collection('rundownPlaylists').insertOne({
+				externalId: playlistExternalId,
+			})
+		}
+
+		const rundownRepository: RundownRepository = mock<RundownRepository>()
+		const mongoDb: MongoDatabase = mock(MongoDatabase)
+		const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+
+		when(rundownRepository.getRundown(rundown.id)).thenReturn(Promise.resolve(rundown))
+		when(mongoDb.getCollection('rundownPlaylists')).thenReturn(db.collection('rundownPlaylists'))
+
+		return new MongoRundownPlaylistRepository(instance(mongoDb), mongoConverter, instance(rundownRepository))
+	}
+
+	async function createTesteeForTwoRundowns(
```

Comment:
You can elimate the need for this method by changing `createTesteeForRundown(includeActiveRundown: boolean, rundownId: string)` to `createTestee(rundowns: Rundown[])`. 

Notice that `includeActivateRundown` is also gone?
Boolean paramteres are a bad practice and we should aim to _also_ use best practices in our test code.
Instead you can just check if the the parsed Rundowns are active, if they are, then you can add the information to the `RundownPlaylists` collection.

Then you can use `createTestee` like this:
```
// Create scenario with an active Rundown
const activeRundown: Rundown = new Rundown({ isActive: true } as RundownInterface)
const testee: RundownRepository = createTestee([activeRundown])

// Create scenario with inactive Rundown
const inactiveRundown: Rundown = new Rundown({ isActive: false } as RundownInterface)
const testee: RundownRepository = createTestee([inactiveRundown])

// Create scenario with both active and inactive Rundown
const testee: RundownRepository = createTestee([activeRundown, inactiveRundown])
```

But I would take it one step further. Right now `createTestee` has two responsibilities: Creating the testee _and_ populating the database. I would split that up so you instead get something like this:
```
const activeRundown: Rundown = new Rundown({ isActive: true } as RundownInterface)
const database: MongoDatabase = populateDatabase([activeRundown])
const testee: RundownRepository = createTestee(database)
```

And while we are at it you can use Dependency Inversion and make the return type `createTestee()` be `RundownRepository` instead of `MongoRundownPlaylistRepository`. Then you are not tempted to make public methods that does not exist on the interface :)

Reply:
I have tried to implement you ideas, let me know. I moved away from `populateDatabase()` but that might only have failed due to the async timers on the tests. Let me know if i should bring it back!